tests/functional/workloads/cnv/test_vm_cloning_ops.py

```diff
@@ -62,3 +62,60 @@ def test_vm_clone(self, cnv_workload, clone_vm_workload, setup_cnv):
             ), f"Failed: MD5 comparison between source {vm_obj.name} and cloned {clone_obj.name} VMs"
             run_dd_io(vm_obj=clone_obj, file_path=file_paths[1])
             clone_obj.stop()
+
+    @workloads
+    @pytest.mark.polarion_id("OCS-6299")
+    def test_vm_snapshot_ops(
+        self, cnv_workload, snapshot_factory, snapshot_restore_factory
+    ):
+        """
+        This test performs the VM PVC snapshot operations
+
+        Test steps:
+        1. Create VMs, add data(e.g., files) to all the VMs
+        2. Create a snapshot for a VM backed pvc
+        3. Restore the snapshot (to same access mode of the parent PVC and storage_class) by following the
+        documented procedure from ODF official docs
+        4. Create new vm using restored pvc Verify existing data of the VM are not changed.
+        5. Add further data(e.g., new file) to the VM
+        6. Repeat the above procedure for all the VMs in the system
+        7. Delete all the VMs created as part of this test
+        """
+        file_paths = ["/file.txt", "/new_file.txt"]
+        # TODO: Add multi_cnv fixture to configure VMs based on specifications
+        vm_obj = cnv_workload(
+            volume_interface=constants.VM_VOLUME_PVC,
+            source_url=constants.CNV_FEDORA_SOURCE,
+        )[0]
+        # Writing IO on source VM
+        source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
+        # Stopping VM before taking snapshot of the VM PVC
+        vm_obj.stop()
```

Comment:
As per doc I guess no need to stop VM.
https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html/managing_and_allocating_storage_resources/managing-persistent-volume-claims_rhodf#expanding-persistent-volume-claims_rhodf